app/src/main/java/ca/llamabagel/transpo/search/ui/viewholders/SearchResult.kt

```diff
@@ -29,7 +30,7 @@ sealed class SearchResult {
     data class CategoryHeader(
         val header: String,
         override val id: String = "",
-        override val type: String = "category"
+        override val type: SearchFilters = SearchFilters.CATEGORY
```

Comment:
This should be overridden in the class body, not the constructor. Technically, this means that a `CategoryHeader` could be initialized with a `type` of `SearchFilters.ROUTE`. This applies to all concrete implementations of `SearchResult`.